src/asm/parser.y

```diff
@@ -849,7 +849,7 @@ redef_id	: T_POP_REDEF {
 		}
 ;
 
-scoped_id	: T_ID | T_LOCAL_ID;
+scoped_id	: T_ID | T_LOCAL_ID | T_LABEL;
```

Comment:
I made this change because `nop::nop` works as expected, and `call Label::ret` should too, but our hacky lexer heuristic detects `Label::` as a `T_LABEL` not a `T_ID`. But that's okay, because treating `T_LABEL` like `T_ID` here is unambiguous and does not conflict with anything.

Reply:
Ack. Hopefully this is also the case in the Rust port. A test case for this is crucial.

Reply:
Yeah, test/asm/multiple-instructions.asm does cover this.

Reply:
That much should be documented in the source itself, actually.

```suggestion
// T_LABEL covers identifiers followed by a colon (e.g. `call Function::ret`,
// to be read as `call Function :: ret`). This should not conflict with anything.
scoped_id	: T_ID | T_LOCAL_ID | T_LABEL;
```